src/shell/signals.c

```diff
@@ -65,6 +99,25 @@ static int signals_init (flux_plugin_t *p,
     return 0;
 }
 
+static int signals_init (flux_plugin_t *p,
+                         const char *topic,
+                         flux_plugin_arg_t *args,
+                         void *data)
+{
+    flux_shell_t *shell = flux_plugin_get_shell (p);
+    if (!shell)
+        return -1;
+
+    set_init_signal_callbacks ();
+
+    if (flux_plugin_add_handler (p,
+                                 "shell.start",
+                                 trap_signals,
+                                 shell) < 0)
+        shell_die_errno (1, "flux_plugin_add_handler");
```

Comment:
Hm, this introduces a small race between when tasks start and the signal watcher gets set up. Are we just creating another problem? Tasks could be started and the shell will just call shell_die() if a signal is received in this window.

Reply:
Yes, it's something I did realize and it perhaps needs a comment somewhere.  Given the potential unknown error that could occur under the current circumstance, I figure "extending the window" of when we catch SIGTERM to just terminate was an acceptable compromise.  It's also b/c the race is pretty rare and unlikely.

Reply:
I should also probably add a note in `signals.c` that `shell_die()` will call `flux_shell_killall()` internally.

Reply:
Makes sense, thanks!